SCClassLibrary/Common/Audio/Delays.sc

```diff
@@ -1,14 +1,33 @@
-Delay1 : PureUGen {
-
-	*ar { arg in = 0.0, mul = 1.0, add = 0.0;
-		^this.multiNew('audio', in).madd(mul, add)
-	}
-	*kr { arg in = 0.0, mul = 1.0, add = 0.0;
-		^this.multiNew('control', in).madd(mul, add)
-	}
+Delay1 : Filter {
+
+    *ar { arg in = 0.0, mul = 1.0, add = 0.0, x1 = 0.0;
+        // nil initializes the delay buffer with the input value
+        x1 = x1 ?? in;
```

Comment:
> should either be x1 = x1 ? in; or x1 = x1 ?? { in };

thanks, updated!

> Well, if you wanted it to match in, you'd have to write nil explicitly. This is quite implicit, I'd actually leave this line 5 out.

I'm not quite sure what you mean. Because the default is `0.0`, `nil` would need to be set explicitly to make `x1` the same as `in` by design—in which case line 5 would be necessary.

Reply:
sorry for my late reply, I had not seen a notification of the update you made.
```supercollider
// so for initialisation with the input
// the intention is that you write:
DelayL.ar(in, 0.5, 0.5, nil)  
// why not write:
DelayL.ar(in, 0.5, 0.5, in)  
```

that would make it more explicit and you don't need the hidden logic in the ugen interface.

Reply:
I see what you mean that intention can be inferred from the default values being explicitly made `nil` in the class method definition. But in fact both are valid as is:
```supercollider
Delay1.ar(in, x1: in) 
Delay1.ar(in, x1: nil) // equivalent but obscure 
```
and I agree your suggestion is preferred for clarity.
The default behavior is more obvious at control rate:
```supercollider
Delay1.kr(in) // default behavior: predelay == in
Delay1.kr(in, x1: 0.0) // explicit predelay value
```
I've removed the explicit `nil` argument values from the method definitions.

I also revisited the help files with fresh eyes, and made changes that hopefully also make the intention more clear.

Reply:
Thank you, I understand better now. If I understand this correctly, unlike at control rate, using `in` as `x1` in an audio rate UGen is not more standard than anything else, right? I read that from your reconstruction example.

Reply:
IIU your question correctly, yes, based on the prior conversation we arrived at the "standard" (default) behavior of **x1** being `0.0`, so specifying `x1: in` would be more exceptional, explicit behavior. I.e. the default follows the "analog" model of delay where silence precedes the delayed signal.

Reply:
OK, good, thank you for the explantion. I would then suggest to remove the `?? { in }` phrases in the audio rate version, because it is unnecessarily implicit to define that via `nil`. In control rate, it is good as it is!

Reply:
> I would then suggest to remove the `?? { in }` phrases in the audio rate version, because it is unnecessarily implicit to define that via `nil`.

In that case, `nil` would no longer be valid for `x1` at audio rate—I can see your rationale for that. I should then update the description in the help docs and perhaps add some clarifying language about this additional difference between ar and kr.

Reply:
yes, that would be good. I think it will be much better …

Reply:
After making the change, I tried to update the description of the behavior in the help files, and it actually came across as more convoluted.

> it is unnecessarily implicit to define that via `nil`
I do see your point with this. However, that logic not visible to the user unless they're inspecting the class file, and in effect supports more input types rather than less.

To try to illustrate this, I updated the help description and added clarifying comments in the class file. Have a look at the latests two commits to see those changes, and let me know if you still think it would be better to remove support for `nil` at audio rate.